hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/s3guard.md

```diff
@@ -1021,6 +1117,41 @@ java.io.IOException: Invalid region specified "iceland-2":
 
 The region specified in `fs.s3a.s3guard.ddb.region` is invalid.
 
+### Error `RemoteFileChangedException`
+
+An exception like the following could occur for a couple of reasons:
+
+* the S3Guard metadata is out of sync with the true S3 metadata.  For
+example, the S3Guard DynamoDB table is tracking a different ETag than the ETag
+shown in the exception.  This may suggest the object was updated in S3 without
+involvement from S3Guard or there was a transient failure when S3Guard tried to
+write to S3.
+
+* S3 is exhibiting read-after-overwrite eventual consistency.  The S3Guard
+metadata was updated with a new ETag during a recent write, but the current read
+is not seeing that ETag due to S3 eventual consistency.  This exception prevents
+the reader from an inconsistent read where the reader sees an older version of
+the file.
+
+```
+org.apache.hadoop.fs.s3a.RemoteFileChangedException: open 's3a://my-bucket/test/file.txt':
+  ETag change reported by S3 while reading at position 0.
+  Version 4e886e26c072fef250cfaf8037675405 was unavailable
```

Comment:
I think there's ETag/Version mismatch here - shouldn't this say "Version change reported...", or "ETag ... was unavailable"?

Reply:
Looks like the "ETag" part is variable but "Version" isn't. Which could be confusing especially if versioning is not enabled, I would suggest to just remove the word "Version" from this error message.

Reply:
This currently comes from:

```
String.format("%s change "
                    + CHANGE_REPORTED_BY_S3
                    + " while reading"
                    + " at position %s."
                    + " Version %s was unavailable",
                getSource(),
                pos,
                getRevisionId()
```

The word "Version" is used generically.  I'll restructure like this:

```
String.format("Version change "
                    + CHANGE_REPORTED_BY_S3
                    + " while reading"
                    + " at position %s."
                    + " %s %s was unavailable",
                pos,
                getSource(),
                getRevisionId()
```

Then,  this message will read like this:

Version change reported by S3 while reading at position 0. ETag 4e886e26c072fef250cfaf8037675405 was unavailable

Reply:
Edit: I'll take out the word "Version" as you suggest so it just reads like this:

Change reported by S3 while reading at position 0. ETag 4e886e26c072fef250cfaf8037675405 was unavailable